### PR TITLE
plan, executor: open the new plan switch for all test.

### PR DESCRIPTION
**Pull request by hanfei1991**
@shenli @zimulala @winoros @lamxTyler @XuHuaiyu  PTAL

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -21,13 +21,15 @@ import (
...
 	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/util/codec"
 	"github.com/pingcap/tidb/util/types"
 	"github.com/pingcap/tipb/go-tipb"
 )
 
-// UseDAGPlanBuilder means we should use new planner and dag pb.
-var UseDAGPlanBuilder = false
+func useDAGPlanBuilder(ctx context.Context) bool {
```
Comment (XuHuaiyu) [inline plan/plan.go]: comment for this func.

**Comment by zimulala** [comment]
Please fix CI.

File: plan/new_physical_plan_builder.go [hunk 2]
```diff
@@ -437,7 +437,17 @@ func (p *DataSource) tryToGetMemTask(prop *requiredProp) (task taskProfile, err
...
+	if len(p.pushedDownConds) > 0 {
+		sel := Selection{
+			Conditions: p.pushedDownConds,
+		}.init(p.allocator, p.ctx)
+		sel.SetSchema(p.schema)
+		sel.SetChildren(memTable)
+		retPlan = sel
+	}
+	task = &rootTaskProfile{p: retPlan}
+
```
Comment (XuHuaiyu) [inline plan/new_physical_plan_builder.go]: remove this blank line

File: executor/executor_test.go [hunk 3]
```diff
@@ -1131,9 +1130,7 @@ func (s *testSuite) TestAdapterStatement(c *C) {
 }
 
 func (s *testSuite) TestPointGet(c *C) {
-	plan.UseDAGPlanBuilder = true
```
Comment (shenli) [inline executor/executor_test.go]: This test will use old planner?

Comment (hanfei1991) [inline executor/executor_test.go]: no

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**